File: demo_python_tf/demo_python_tf/Dynamic_tf_broadcaster.py
import rclpy
from rclpy.node import Node
from tf2_ros import TransformBroadcaster
from geometry_msgs.msg import TransformStamped
from tf_transformations import quaternion_from_euler
import math 


class TFBroadcaster(Node):
    
    def __init__(self):
        super().__init__('tf_broadcaster')
        self.broadcaster = TransformBroadcaster(self)
        self.timer_ = self.create_timer(0.01,self.publish_tf)
        
    def publish_tf(self):
        
        """
        Broadcast coordinate relationship from camera_link to bottom_link
        """
        
        tranform = TransformStamped()
        tranform.header.frame_id = 'camera_link'
        tranform.child_frame_id = 'bottom_link'
        tranform.header.stamp = self.get_clock().now().to_msg()
        
        tranform.transform.translation.x = 0.2
        tranform.transform.translation.y = 0.3
        tranform.transform.translation.z = 0.5
        
        q = quaternion_from_euler(0,0,0)

        tranform.transform.rotation.x = q[0]
        tranform.transform.rotation.y = q[1]
        tranform.transform.rotation.z = q[2]
        tranform.transform.rotation.w = q[3]
        
        self.broadcaster.sendTransform(tranform)
        self.get_logger().info(f"Broadcast tf: {tranform}")
        

def main():
    
    rclpy.init()
    node = TFBroadcaster()
    rclpy.spin(node)
    rclpy.shutdown()
    

File: demo_cpp_tf/src/dynamic_tf_broadcaster.cpp
#include "geometry_msgs/msg/transform_stamped.hpp"
#include "rclcpp/rclcpp.hpp"
#include "tf2/LinearMath/Quaternion.h"
#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
#include "tf2_ros/transform_broadcaster.h"
#include "chrono"

using namespace std::chrono_literals;

class TFBroadcaster: public rclcpp::Node
{
private:
   std::shared_ptr<tf2_ros::TransformBroadcaster> broadcaster_;
   rclcpp::TimerBase::SharedPtr timer_;


public:
    TFBroadcaster():Node("tf_broadcaster"){
        broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(this);
        this->publish_tf();
        timer_ = this->create_wall_timer(100ms,std::bind(&TFBroadcaster::publish_tf,this));
    }
    
    void publish_tf(){
        geometry_msgs::msg::TransformStamped transform;
        transform.header.stamp = this->get_clock()->now();
        transform.header.frame_id = "map";
        transform.child_frame_id = "base_link";
        transform.transform.translation.x = 1.0;
        transform.transform.translation.y= 2.0;
        transform.transform.translation.z = 3.0;
        tf2::Quaternion q;
        q.setRPY(0.0,0.0,30/180.0*M_PI);
        transform.transform.rotation = tf2::toMsg(q);
        this->broadcaster_->sendTransform(transform);
    }
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<TFBroadcaster>();
    rclcpp::spin(node);
    return 0;
}

